ravenframework/SupervisedLearning/DMD/DynamicModeDecompositionControl.py

```diff
@@ -199,6 +236,43 @@ def _handleInput(self, paramInput):
       if str(self.parametersIDs[i]).endswith('_init'):
         self.parametersIDs.remove(self.parametersIDs[i])
 
+  def initializeModel(self, dmdParams):
+    """
+      Method to initialize the surrogate model with a dmdParams dictionary
+      @ In, dmdParams, dict, the dictionary containin the parameters/settings to instanciate the model
+      @ Out, None
+    """
+    pass
+
+  def _getTimeScale(self,dmd=True):
+    """
+      Get the ts of the dmd (if dmd = True) or training (if dmd = False) reconstructed time scale.
+      @ In, dmd, bool, optional, True if dmd time scale needs to be returned, othewise training one
+      @ Out, timeScale, numpy.array, the dmd or training reconstructed time scale
+    """
+    timeScaleInfo = self.timeScales['dmd'] if dmd else self.timeScales['training']
+    timeScale = np.arange(timeScaleInfo['t0'], (timeScaleInfo['intervals']+1)*timeScaleInfo['dt'], timeScaleInfo['dt'])
+    return timeScale
```

Comment:
Can this method combined with DMDBase method?

Reply:
This is needed to avoid conflicts with the DynamicModeDecompositionControl (that is used by FARM external plugin). Maybe once this is merged, ANL can be informed to check how the DynamicModeDecompositionControl (now also available in PyDMD) can be recast.